mailbox/event/event-memory/src/main/java/org/apache/james/mailbox/events/delivery/InVmEventDelivery.java

```diff
@@ -51,55 +54,64 @@
 
     private final MetricFactory metricFactory;
     private final RetryBackoffConfiguration retryBackoff;
+    private final EventDeadLetters eventDeadLetters;
 
     @Inject
     @VisibleForTesting
-    public InVmEventDelivery(MetricFactory metricFactory, RetryBackoffConfiguration retryBackoff) {
+    public InVmEventDelivery(MetricFactory metricFactory, RetryBackoffConfiguration retryBackoff, EventDeadLetters eventDeadLetters) {
         this.metricFactory = metricFactory;
         this.retryBackoff = retryBackoff;
+        this.eventDeadLetters = eventDeadLetters;
     }
 
+    @VisibleForTesting
     public InVmEventDelivery(MetricFactory metricFactory) {
-        this(metricFactory, RetryBackoffConfiguration.DEFAULT);
+        this(metricFactory, RetryBackoffConfiguration.DEFAULT, new MemoryEventDeadLetters());
     }
 
     @Override
     public ExecutionStages deliver(Collection<MailboxListener> mailboxListeners, Event event) {
```

Comment:
Maybe a single deliver method can do the work without the `DeliverableListener` that from your own point of view is badly named.

A badly named concept is most probably also a bad concept.

What I propose is to strengthen DeliveryOptions:

```
DeliveryOptions.withRetry()
    .saveUponFailureInDeadLetter(group)
    .build()
````

Or

```
DeliveryOptions.none()
```

Then 

Introduce a `deliver(MailboxListener listener, Event event, DeliveryOptions options)`

Reply:
We should also be able to easily combine `ExecutionStages` together so that the caller could easily be reducing upon them.